api/ResponseTypes.d.ts

```diff
@@ -0,0 +1,132 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+interface SearchResponse<TSource = unknown, TAggregations = unknown> {
+  took: number
+  timed_out: boolean
+  _scroll_id?: string
+  _shards: Shards
+  hits: {
+    total: {
+      value: number
+      relation: string
+    }
+    max_score: number
+    hits: Array<{
+      _index: string
+      _type: string
+      _id: string
+      _score: number
+      _source: TSource
+      _version?: number
+      _explanation?: Explanation
+      fields?: Record<string, unknown>
+      highlight?: Record<string, string[]>
+      inner_hits?: Record<string, unknown>
+      matched_queries?: string[]
+      sort?: any[]
+    }>
+  }
+  aggregations?: TAggregations
+}
+
+interface Shards {
+  total: number
+  successful: number
+  failed: number
+  skipped: number
+}
+
+interface Explanation {
+  value: number
+  description: string
+  details: Explanation[]
+}
+
+interface MSearchResponse<TSource = unknown, TAggregations = unknown> {
+  responses: Array<SearchResponse<TSource, TAggregations>>
+}
+
+interface CreateResponse {
+  _shards: Shards
+  _index: string
+  _type: string
+  _id: string
+  _version: number
+  _seq_no: number
+  _primary_term: number
+  result: string
+}
+
+interface IndexResponse extends CreateResponse {}
+
+interface DeleteResponse {
+  _shards: Shards
+  _index: string
+  _type: string
+  _id: string
+  _version: number
+  _seq_no: number
+  _primary_term: number
+  result: string
+}
+
+interface UpdateResponse {
+  _shards: Shards
+  _index: string
+  _type: string
+  _id: string
+  _version: number
+  result: string
+}
```

Comment:
Elasticsearch 7 supports returning the document source in update requests, e.g. the following request
```json
POST /<index>/_update/<id>
{
  "_source": "*",
  "doc": {
    "field": "val"
  }
}
```
gives the following response
```json
{
  "_index" : "<index>",
  "_type" : "_doc",
  "_id" : "<id>",
  "_version" : 5,
  "result" : "updated",
  "_shards" : {
    "total" : 2,
    "successful" : 1,
    "failed" : 0
  },
  "_seq_no" : 53,
  "_primary_term" : 1,
  "get" : {
    "_seq_no" : 53,
    "_primary_term" : 1,
    "found" : true,
    "_source" : {
      "field": "val"
    }
  }
}
```
So the user should be able to specify the type of that `_source` (furthermore this `get` fields should be an optional property in `UpdateResponse<T>`.


@delvedor also note that the shards object I got from Elasticsearch doesn't contain `skipped` fields, this is either a bug or an intended behavior, so if the latter we should mark it as optional: `interface Shards { skipped?: number }`

Also note that there are two more fields:
```json
{
  "_seq_no" : 53,
  "_primary_term" : 1
}
```
I think they also should be included in `interface UpdateResponse`